bciers/libs/utils/src/sessionRoleContext.tsx

```diff
@@ -0,0 +1,21 @@
+"use client";
+import {
+  AwaitedReactNode,
+  createContext,
+  JSXElementConstructor,
+  ReactElement,
+  ReactNode,
+  ReactPortal,
+} from "react";
+
+export const SessionRoleContext = createContext(undefined);
+
+const SessionRoleContextProvider = (props: any) => {
+  return (
+    <SessionRoleContext.Provider value={props.value}>
+      {props.children}
+    </SessionRoleContext.Provider>
+  );
+};
```

Comment:
we can probably type this a bit better:
```suggestion

export const SessionRoleContext = createContext<string | undefined>(undefined);

interface SessionRoleContextProviderProps {
  value: string | undefined;
  children: React.ReactNode;
}

const SessionRoleContextProvider: React.FC<SessionRoleContextProviderProps> = (props) => {
  return (
    <SessionRoleContext.Provider value={props.value}>
      {props.children}
    </SessionRoleContext.Provider>
  );
};
```